src/main/java/projectbookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemDto> findAllOrderItemsByOrderId(
+            Pageable pageable, Long userId, Long orderId) {
+        Order order = getExistedOrder(orderId, userId);
+        checkOrderBelongsUser(userId, order);
+        return order.getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemDto findOrderItemById(Long userId, Long itemId) {
+        return orderItemMapper.toDto(
+                orderItemRepository.findByIdAndUserId(itemId, userId).orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "Can`t find an orderItem by orderItem id: " + itemId))
+        );
+    }
+
+    @Override
+    public OrderDto update(Long id, User user, UpdateOrderStatusDto dto) {
+        Order order = getExistedOrder(id, user.getId());
+        order.setStatus(dto.getStatus());
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    private BigDecimal calculateTotalPrice(Set<CartItem> cartItems) {
+        return cartItems.stream()
+                .map(item -> item.getBook().getPrice()
+                        .multiply(BigDecimal.valueOf(item.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private Set<OrderItem> createOrderItems(Order order, ShoppingCart shoppingCart) {
+        Set<OrderItem> orderItems = shoppingCart.getCartItems().stream()
+                .map(item -> createOrderItem(item, order))
+                .collect(Collectors.toSet());
+        orderItemRepository.saveAll(orderItems);
+        return orderItems;
+    }
+
+    private OrderItem createOrderItem(CartItem cartItem, Order order) {
+        OrderItem orderItem = new OrderItem();
+        orderItem.setOrder(order);
+        orderItem.setBook(cartItem.getBook());
+        orderItem.setQuantity(cartItem.getQuantity());
+        orderItem.setPrice(cartItem.getBook().getPrice()
+                .multiply(BigDecimal.valueOf(cartItem.getQuantity())));
+        return orderItem;
+    }
+
+    private Order createNewOrder(CreateOrderRequestDto requestDto,
+                                 User user,
+                                 ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Order.Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(requestDto.getShippingAddress());
+        BigDecimal total = calculateTotalPrice(shoppingCart.getCartItems());
+        order.setTotal(total);
+        return order;
+    }
+
+    private void checkOrderBelongsUser(Long userId, Order order) {
+        if (! order.getUser().getId().equals(userId)) {
+            throw new EntityNotFoundException("Can't find an order by order id: "
+                    + order.getId() + " for user id: " + userId);
```

Comment:
update ex message so that user would know that this order does not belong to him, since that is what we are checking in that method